src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,321 @@
+        let matchedTrials: Array<IDetailedTrialMatch> = [];
+        _.forEach(groupByTrialId, (trialGroup, trialId) => {
+            let matchedTrial:any = _.find( this.props.trials, { 'nctId': trialId } );
+            matchedTrial['priority'] = 0; // highest priority
+            matchedTrial['matches'] = [];
+            const groupByArm = _.groupBy(trialGroup, 'armDescription');
+            _.forEach(groupByArm, (armGroup, armDescription) => {
+                let armMatch: IArmMatch = {
+                    armDescription: armDescription,
+                    drugs: [],
+                    matches: []
+                };
+                const isControlArm = getDrugsAndExcludeControlArms(armDescription, matchedTrial['treatmentList']['step'][0]['arm'], armMatch);
+                if (isControlArm) return; // skip control arm
+
+                const groupByAge = _.groupBy(armGroup, 'trialAgeNumerical');
+                _.forEach(groupByAge, (ageGroup, age) => {
+                    const cancerTypes =  _.uniq(_.map(ageGroup, 'trialOncotreePrimaryDiagnosis'));
+                    let generalCancerTypes: Array<string> = [];
+                    let notCancerTypes: Array<string> = [];
+                    _.map(cancerTypes, (item) => {
+                       if (item.indexOf('!') > -1) {
+                           notCancerTypes.push(item.replace('!', ''));
+                       } else {
+                           generalCancerTypes.push(item);
+                       }
+                    });
+                    let clinicalGroupMatch: IClinicalGroupMatch = {
+                        trialAgeNumerical: age,
+                        trialOncotreePrimaryDiagnosis: {
+                            general: generalCancerTypes,
+                            not: notCancerTypes
+                        },
+                        matches: [],
+                        notMatches: []
+                    };
+                    const groupByGenomicAlteration = _.groupBy(groupByAge[age], 'genomicAlteration');
+                    _.forEach(groupByGenomicAlteration, (genomicAlterationGroup, genomicAlteration) => {
+                        const groupByPatientGenomic = _.groupBy(genomicAlterationGroup, 'patientGenomic');
+                        let genomicGroupMatch: IGenomicGroupMatch = {
+                            genomicAlteration: genomicAlteration,
+                            matches: []
+                        };
+                        _.forEach(groupByPatientGenomic, (patientGenomicGroup) => {
+                            let genomicMatch: IGenomicMatch = {
+                                trueHugoSymbol: patientGenomicGroup[0].trueHugoSymbol,
+                                trueProteinChange: patientGenomicGroup[0].trueProteinChange,
+                                sampleIds: []
+                            };
+                            const sampleIds = _.uniq(_.map(patientGenomicGroup, 'sampleId'));
```

Comment:
shortcut

Reply:
Changed to `const sampleIds = _.uniq(_.map(patientGenomicGroup, (trial: ITrialMatch) => trial.sampleId));`.